cluster/store.go

```diff
@@ -217,12 +219,20 @@ func (st *Store) Open(ctx context.Context) (err error) {
 	st.log.WithFields(logrus.Fields{
 		"name":                 st.cfg.NodeID,
 		"metadata_only_voters": st.cfg.MetadataOnlyVoters,
+		"bootstrap_expect":     st.cfg.BootstrapExpect,
 	}).Info("construct a new raft node")
-	st.raft, err = raft.NewRaft(st.raftConfig(), st, logCache, st.logStore, st.snapshotStore, st.raftTransport)
+
+	st.raft, err = raft.NewRaft(st.raftConfig(), st, st.logCache, st.logStore, st.snapshotStore, st.raftTransport)
 	if err != nil {
 		return fmt.Errorf("raft.NewRaft %v %w", st.raftTransport.LocalAddr(), err)
 	}
 
+	if st.cfg.BootstrapExpect == 1 {
```

Comment:
Question: Could the issue also appear if we have a 3 node cluster -> they all go down and they all change IP ?

Or would a 3 node cluster have `NODE_ID` specified and therefore our usual recovery logic would handle it ?